From 305a9ad1cad5cfd13ff3914160efa5d5c81e2198 Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Thu, 22 Mar 2018 16:14:28 +0800
Subject: [PATCH] arm64: dts: rockchip: Add mac node for rk3308-evb

According to the hardware, the mac is not active, so
disable it's status.

Change-Id: I59d6f6232529de3945ce1c766bb7a3c325bdfa8e
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 .../boot/dts/rockchip/rk3308-evb-v10.dtsi     | 20 +++++++++++++++++++
 arch/arm64/boot/dts/rockchip/rk3308.dtsi      |  7 +++++++
 2 files changed, 27 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
index d578da72bd84..89f6725391b0 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308-evb-v10.dtsi
@@ -177,6 +177,13 @@
 		regulator-max-microvolt = <1800000>;
 	};
 
+	vcc_phy: vcc-phy-regulator {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc_phy";
+		regulator-always-on;
+		regulator-boot-on;
+	};
+
 	vbus_host: vbus-host-regulator {
 		compatible = "regulator-fixed";
 		enable-active-high;
@@ -225,6 +232,19 @@
 	status = "okay";
 };
 
+&mac {
+	phy-supply = <&vcc_phy>;
+	assigned-clocks = <&cru SCLK_MAC>;
+	assigned-clock-parents = <&mac_clkin>;
+	clock_in_out = "input";
+	pinctrl-names = "default";
+	pinctrl-0 = <&rmii_pins &mac_refclk>;
+	snps,reset-gpio = <&gpio0 RK_PB2 GPIO_ACTIVE_LOW>;
+	snps,reset-active-low;
+	snps,reset-delays-us = <0 50000 50000>;
+	status = "disable";
+};
+
 &io_domains {
 	status = "okay";
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3308.dtsi b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
index 0ac6a256a76c..8be66659c587 100644
--- a/arch/arm64/boot/dts/rockchip/rk3308.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3308.dtsi
@@ -143,6 +143,13 @@
 		status = "disabled";
 	};
 
+	mac_clkin: external-mac-clock {
+		compatible = "fixed-clock";
+		clock-frequency = <50000000>;
+		clock-output-names = "mac_clkin";
+		#clock-cells = <0>;
+	};
+
 	psci {
 		compatible = "arm,psci-1.0";
 		method = "smc";
-- 
2.35.3

